src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -333,6 +334,12 @@ public static class EndpointRouteBuilderExtensions
                 }
             }
 
+            // Add delegate attributes as metadata
+            foreach (var attribute in requestDelegate.Method.GetCustomAttributes())
```

Comment:
What is the `Method` for a lambda expression?

Reply:
The compiler always generates a method. You just can't apply attributes to it because you don't have a place to put them.